src/app/utils.ts

```diff
@@ -130,7 +127,8 @@ export default class Utils {
     };
     const key = Object.keys(timeWindows)[index];
     const value = Object.values(timeWindows)[index];
-    return { key, value };
+    const size = Object.keys(timeWindows).length;
+    return { key, value, size };
```

Comment:
May be we should take const timeWindows out, and instead of returning size  here,  add a function getTimeWindowsSize to return that.

Reply:
It does not make any difference, we can do it in anyway.